### DISCUSSION TITLE
在使用CREATE-TABLE-AS-SELECT中的字段，不支持base表中的range和hash字段吗？

### DISCUSSION THREAD

**Discussion by SamuelBin** [apache/doris#12758] [A - General / Q&A]
在使用CREATE-TABLE-AS-SELECT中的字段，不支持base表中的range和hash字段吗？
base表：
CREATE TABLE `t1`(
  `dt` Date,
  `name` VARCHAR(24),
  `description` VARCHAR(64)
) DUPLICATE KEY(dt,name,description) PARTITION BY RANGE (dt) () DISTRIBUTED BY HASH(name) BUCKETS 3 PROPERTIES (
  "dynamic_partition.create_history_partition" = "true",
  "dynamic_partition.time_unit" = "DAY",
  "dynamic_partition.history_partition_num" = "30",
  "dynamic_partition.end" = "7",
  "dynamic_partition.prefix" = "p",
  "dynamic_partition.buckets" = "3",
  "dynamic_partition.replication_num" = "1"
);

create table t2(
  dt,
  name,
  description
) PROPERTIES("replication_num" = "1") as
select
  dt,
  name,
  description,
from
  t1
where
  name= 'xxx';

**Comment by LemonLiTree** [comment]
这个是需要手动指定的，如果按你的语句建会自动分配分区分桶字段(不建议这么做)。
